src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -2619,6 +2619,22 @@ public static unsafe int SequenceCompareTo<T>(this ReadOnlySpan<T> span, ReadOnl
                     valueLength);
         }
 
+        /// <summary>
+        /// Determines whether the specified value appears at the start of the span.
+        /// </summary>
+        /// <param name="span">The span to search.</param>
+        /// <param name="value">The value to compare.</param>
+        public static bool StartsWith<T>(this ReadOnlySpan<T> span, T value) where T : IEquatable<T>? =>
+            span.Length != 0 && (span[0]?.Equals(value) ?? (object?)value is null);
+
+        /// <summary>
+        /// Determines whether the specified value appears at the end of the span.
+        /// </summary>
+        /// <param name="span">The span to search.</param>
+        /// <param name="value">The value to compare.</param>
+        public static bool EndsWith<T>(this ReadOnlySpan<T> span, T value) where T : IEquatable<T>? =>
+            span.Length != 0 && (span[^1]?.Equals(value) ?? (object?)value is null);
```

Comment:
Seems like the JIT might be having a little trouble optimizing this? [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYm5JE1IMhDAN40GlpruYHgECJIYBJXAEUOEDDAAm5ABRsMNg+APJckgCesmLYXAA8QgAW2FAAfAy4sVwAlAwAvGkWViVZcSwAolw+uADqvBhJAQDkAETNORrUJcWWOnp2Dk6uHl6+pIHBYRHR2Ykp6ZnZeYW9JUvlVTX1jRNtHV093Vb9tgz2ji7unt4+2pMh4VExcfOpGWW5BUXH6xtcLAAMjAuABzRoMACE+QYjAAZHD/gBtAB65AAupUAI4cbCSXD+fadNYk37WAbnIYMLZ1BpNeIAFTS/kavFwDCCjxmLwSTP+aAYDIYCjxHBgeQA7kkZDBBQwQC4KjjsBhsMBpIy0gB+b5rUrZIEg8FJKEw+GI/yfVEYrXY3H4/wiyRivJanX+CDAABWMDAGC1OSdYoYbIYXA4kkkxLJayRAFkYI0ID5nOJJP4E0mU2nQmI+BAuLgWABBUGg2C4XC8BQwZwRXhcBugnLotanfSUy40nZJUiallJUOc6bPOZ8z4CoVB8UMKUyuUK5xK3Gq9UwTU61Zkko+GAAM2wEYw/gZeVDXC8YYjTi1er+DEtBuBYIh0NhDARyLRmOXeIJ05yPIQDvP5PkNF8TTfc0Hytb9bV/B0AIYN0H09H0/QDacQ3ZcNI2jABfIA)

Reply:
Should have improved with #102518 / #102808.
I can double-check what it's doing now.

@MihuBot -fast

Reply:
It's generating the same ASM these now
```c#
[Benchmark]
public bool Old()
{
    ReadOnlySpan<char> span = Value;
    return span.Length != 0 && span[0] == '"' && span[^1] == '"';
}

[Benchmark]
public bool ExtGeneric()
{
    ReadOnlySpan<char> span = Value;
    return span.StartsWith('"') && span.EndsWith('"');
}

[Benchmark]
public bool ExtChar()
{
    ReadOnlySpan<char> span = Value;
    return span.StartsWithChar('"') && span.EndsWithChar('"');
}
```